src/components/checkbox/index.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { forwardRef, ReactNode } from 'react';
+import {
+  BoxProps,
+  Checkbox as ChakraCheckbox,
+  RecipeVariantProps,
+  SwitchCheckedChangeDetails,
+  useSlotRecipe,
+} from '@chakra-ui/react';
+
+import { checkboxRecipe } from 'src/themes/shared/slotRecipes/checkbox';
+
+type CheckboxVariantProps = RecipeVariantProps<typeof checkboxRecipe>;
+
+export type CheckboxProps = CheckboxVariantProps & {
+  name: string;
+  value?: string;
+  disabled?: boolean;
+  checked?: boolean;
+  invalid?: boolean;
+  indeterminate?: boolean;
+  readOnly?: boolean;
+  fullWidth?: boolean;
+  label?: ReactNode | string;
+  paddingY?: BoxProps['paddingY'];
+  fontWeight?: 'regular' | 'bold';
+  variant?: 'alignCenter' | 'alignTop' | 'alignTopForSmallSize';
```

Comment:
We can pick the variants for the recipe https://www.chakra-ui.com/docs/theming/recipes#typescript